sdk/python/kfp/components/base_component.py

```diff
@@ -93,9 +93,26 @@ def __call__(self, *args, **kwargs) -> pipeline_task.PipelineTask:
     @property
     def pipeline_spec(self) -> pipeline_spec_pb2.PipelineSpec:
         """Returns the pipeline spec of the component."""
-        return self.component_spec.to_pipeline_spec()
+        with BlockPipelineTaskRegistration():
+            return self.component_spec.to_pipeline_spec()
 
     @abc.abstractmethod
     def execute(self, **kwargs):
         """Executes the component locally if implemented by the inheriting
         subclass."""
+
+
+class BlockPipelineTaskRegistration:
+    """Temporarily stop registering tasks to the default pipeline.
```

Comment:
Out of curiosity, is there a plan to remove this "hack"? If so, consider adding a TODO comment.

Reply:
I will add a TODO, but I don't have any immediate plans to remove this. It is indeed a bit hacky, but possibly a necessary evil since the compiler is largely built on top of the idea of a "default pipeline", but this handles the special case of a compiled component, which has no concept of a default pipeline.

Reply:
That sounds fine.